{
    "source": "BDSA",
    "name": "BDSA-2019-4233",
    "title": "Python Vulnerable to Denial-of-service (DoS) via Resource Consumption in 'Lib/zipfile.py' Library",
    "description": "It was discovered that python contains a denial-of-service (DoS) vulnerability. An attacker could exploit this flaw using a zip bomb to trigger a DoS condition via excessive resource consumption.",
    "technicalDescription": "The vulnerability occurs due to insufficient ZIP bomb detection in the `zipfile.py` library. This could allow an attacker to cause a DoS condition via ZIP bomb attack.\n\n**Note** The vendor opted to update the documentation with a warning to the application developers to limit resource consumption.",
    "publishedDate": "2020-02-05T15:07:07.770Z",
    "updatedDate": "2021-07-02T13:33:21.983Z",
    "disclosureDate": "2019-03-28T00:00:00.000Z",
    "solution": "Fixed in [**3.9.0a1**](https://github.com/python/cpython/releases/tag/v3.9.0a1) by [this](https://github.com/python/cpython/commit/3ba51d587f6897a45301ce9126300c14fcd4eba2) commit.\n\nFixed in [**3.8.0rc1**](https://github.com/python/cpython/releases/tag/v3.8.0rc1) by [this](https://github.com/python/cpython/commit/c5a672315d) commit.",
    "severity": "MEDIUM",
    "cvss2": {
        "baseScore": 5.0,
        "impactSubscore": 2.9,
        "exploitabilitySubscore": 10.0,
        "severity": "LOW",
        "accessVector": "NETWORK",
        "accessComplexity": "LOW",
        "authentication": "NONE",
        "confidentialityImpact": "NONE",
        "integrityImpact": "NONE",
        "availabilityImpact": "PARTIAL",
        "temporalMetrics": {
            "exploitability": "UNPROVEN",
            "remediationLevel": "OFFICIAL_FIX",
            "reportConfidence": "CONFIRMED",
            "score": 3.7
        },
        "vector": "(AV:N/AC:L/Au:N/C:N/I:N/A:P/E:U/RL:OF/RC:C)"
    },
    "cvss3": {
        "baseScore": 7.5,
        "impactSubscore": 3.6,
        "exploitabilitySubscore": 3.9,
        "severity": "MEDIUM",
        "attackVector": "NETWORK",
        "attackComplexity": "LOW",
        "confidentialityImpact": "NONE",
        "integrityImpact": "NONE",
        "availabilityImpact": "HIGH",
        "privilegesRequired": "NONE",
        "scope": "UNCHANGED",
        "userInteraction": "NONE",
        "temporalMetrics": {
            "exploitability": "UNPROVEN",
            "remediationLevel": "OFFICIAL_FIX",
            "reportConfidence": "CONFIRMED",
            "score": 6.5
        },
        "vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H/E:U/RL:O/RC:C"
    },
    "useCvss3": true,
    "classifications": [],
    "zeroDay": false,
    "underReview": false,
    "parentAdvisory": false,
    "credit": "JUN-WEI SONG",
    "workaround": "",
    "vendorFixDate": "2019-09-11T00:00:00.000Z",
    "bdsaTags": [],
    "overallScore": 6.5,
    "_meta": {
        "allow": [],
        "href": "https://trendmicro.app.blackduck.com/api/vulnerabilities/BDSA-2019-4233",
        "links": [
            {
                "rel": "cwes",
                "href": "https://trendmicro.app.blackduck.com/api/cwes/CWE-409"
            },
            {
                "rel": "related-vulnerability",
                "href": "https://trendmicro.app.blackduck.com/api/vulnerabilities/CVE-2019-9674",
                "label": "NVD"
            },
            {
                "rel": "bdsa-ranges",
                "href": "https://trendmicro.app.blackduck.com/api/vulnerabilities/BDSA-2019-4233/ranges"
            },
            {
                "rel": "reference",
                "href": "https://bugs.python.org/issue36260",
                "label": "ADVISORY"
            },
            {
                "rel": "reference",
                "href": "https://github.com/python/cpython/commit/3ba51d587f6897a45301ce9126300c14fcd4eba2",
                "label": "PATCH"
            },
            {
                "rel": "reference",
                "href": "https://github.com/python/cpython/commit/c5a672315d",
                "label": "PATCH"
            },
            {
                "rel": "reference",
                "href": "https://github.com/python/cpython/releases/tag/v3.8.0rc1",
                "label": "VENDOR_UPGRADE"
            },
            {
                "rel": "reference",
                "href": "https://github.com/python/cpython/releases/tag/v3.9.0a1",
                "label": "VENDOR_UPGRADE"
            }
        ]
    }
}